### PR TITLE
Add base lockfiles to build_and_upload_dependencies.sh

### PR DESCRIPTION
**Pull request by beckerhe**
This makes use of base lockfiles in our dependency build script.

Furthermore this fixes a regression in `build.*` which used to build dependencies when not available as binary prebuilts. This is not relevant for the team, but for external contributors using the public remotes which don't have all the prebuilts.

### CODE REVIEW

File: build.ps1 [hunk 1]
```diff
@@ -49,6 +49,7 @@ foreach ($profile in $profiles) {
   New-Item -ItemType Directory -Force -Path ".\build_$profile\" | Out-Null
 
   & $conan.Path lock create "$PSScriptRoot\conanfile.py" --user=orbitdeps --channel=stable `
+    --build outdated `
```
Comment (IrinaShkviro) [inline build.ps1:52]: Maybe we could you
```suggestion
    --build=outdated `
```
as below for consistency? (and the same in `build.sh`)

**Review by IrinaShkviro — APPROVED** [review]
lgtm, thanks!

File: build.ps1 (hunk 1 above)
Comment (beckerhe) [reply build.ps1:52]: Good point. Will fix that!

**Comment by beckerhe** [comment]
So this was more complicated as expected.

The "cmake_installer not in lockfile" error might be a bug in conan or we do something wrong.

But it seems that we can fix it by applying a change I wanted to do for a long time.

The commit messages have more information on the details

@IrinaShkviro PTAL
@danielfenner Adding you as a reviewer.

**Review by danielfenner — APPROVED** [review]
Thanks, lgtm.

**Review by IrinaShkviro — APPROVED** [review]

**Pull request merged**